### PR TITLE
feat: add prompt to emitter

### PR DESCRIPTION
**Pull request by dariuszd21**
- [ ] Have you followed the guidelines for contributing?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `tox`?

Add a functionality to prompt user for an input.
Additionally with `hide` set to `True` user may be asked for a secret input.

(CRAFT-3739)

### CODE REVIEW

File: craft_cli/messages.py [hunk 1]
```diff
@@ -806,6 +807,27 @@ def confirm(self, prompt: str, *, default: bool = False) -> bool:
             return False
         return default
 
+    @_active_guard()
+    def prompt(self, prompt: str, *, hide: bool = False) -> str:
+        """Prompt user for input.
+
+        If stdin is not a tty a CraftError is raised.
+
```
Comment (bepri) [inline craft_cli/messages.py:815]: Need to document the `prompt` parameter

File: craft_cli/messages.py [hunk 2]
```diff
@@ -806,6 +807,27 @@ def confirm(self, prompt: str, *, default: bool = False) -> bool:
...
+        :param hide: hide user input if True.
+        :returns: value that was provided by user.
+        :raises: CraftError if shell is not interactive or input is empty
```
Comment (bepri) [inline craft_cli/messages.py:818]:
```suggestion
        :raises: CraftError if shell is not interactive or input is empty.
```

**Review by bepri — APPROVED** [review]
Looks good! Just nitpicks, so I'm approving :)

File: craft_cli/messages.py (hunk 1 above)
Comment (dariuszd21) [reply craft_cli/messages.py:815]: Added in https://github.com/canonical/craft-cli/pull/315/commits/2a95dc594636024e7702aa70b84c8dd9877111ea

File: craft_cli/messages.py (hunk 2 above)
Comment (dariuszd21) [reply craft_cli/messages.py]: Done in https://github.com/canonical/craft-cli/pull/315/commits/2a95dc594636024e7702aa70b84c8dd9877111ea

**Review by tigarmo — APPROVED** [review]
thanks!

**Pull request merged**